db/cassandra_client.go

```diff
@@ -76,90 +76,12 @@ type PenetrationMetrics struct {
 }
 
 func NewCassandraClient(conf *configuration.Config, testOnly bool) (*CassandraClient, error) {
-	// init
-	hosts := conf.GetStringList("xconfwebconfig.database.hosts")
-	cluster := gocql.NewCluster(hosts...)
-
-	cluster.Consistency = gocql.LocalQuorum
-	cluster.ProtoVersion = int(conf.GetInt32("xconfwebconfig.database.protocolversion", ProtocolVersion))
-	cluster.DisableInitialHostLookup = DisableInitialHostLookup
-	cluster.Timeout = time.Duration(conf.GetInt32("xconfwebconfig.database.timeout_in_sec", 1)) * time.Second
-	cluster.ConnectTimeout = time.Duration(conf.GetInt32("xconfwebconfig.database.connect_timeout_in_sec", 1)) * time.Second
-	cluster.NumConns = int(conf.GetInt32("xconfwebconfig.database.connections", DefaultConnections))
-
-	cluster.RetryPolicy = &gocql.DowngradingConsistencyRetryPolicy{
-		[]gocql.Consistency{
-			gocql.LocalQuorum,
-			gocql.LocalOne,
-			gocql.One,
-		},
-	}
-
-	localDc := conf.GetString("xconfwebconfig.database.local_dc")
-	if len(localDc) > 0 {
-		cluster.PoolConfig.HostSelectionPolicy = gocql.DCAwareRoundRobinPolicy(localDc)
-	}
-
-	user := conf.GetString("xconfwebconfig.database.user")
-	encryptedPassword := conf.GetString("xconfwebconfig.database.encrypted_password")
-	isSslEnabled := conf.GetBoolean("xconfwebconfig.database.is_ssl_enabled")
-
-	//build codec
-	codec := security.NewAesCodec()
-
-	var password string
-	var err error
-
-	if encryptedPassword != "" {
-		password, err = codec.Decrypt(encryptedPassword)
-		if err != nil {
-			log.Error(err.Error())
-			return nil, err
-		}
-	} else {
-		password = conf.GetString("xconfwebconfig.database.password")
-	}
-
-	cluster.Authenticator = gocql.PasswordAuthenticator{
-		Username: user,
-		Password: password,
-	}
-
-	if isSslEnabled {
-		sslOpts := &gocql.SslOptions{
-			EnableHostVerification: false,
-		}
-		cluster.SslOpts = sslOpts
-	}
-
-	// Use the appropriate keyspace
-	var deviceKeyspace string
-	if testOnly {
-		cluster.Keyspace = conf.GetString("xconfwebconfig.database.test_keyspace", DefaultTestKeyspace)
-		deviceKeyspace = conf.GetString("webconfig.database.device_test_keyspace", DefaultDeviceTestKeyspace)
+	isIamBasedAccessEnabled := conf.GetBoolean("xconfwebconfig.database.iam_based_access_enabled")
+	if isIamBasedAccessEnabled {
```

Comment:
if someone want to access the AWS keyspace with AccessKey and SecretKey then there is no provision as of now. Can you enhance the same?

Reply:
I have enhanced the same.